src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +112,872 @@ static DamageTable D_80AD2550 = {
+                (gSaveContext.entranceIndex == 0xC060)) {
+                this->unk_1EA = (u16)(this->unk_1EA | 1);
+            }
+            this->unk_1F0 = 1;
+            if (globalCtx->sceneNum == 0x63) {
+                if ((gSaveContext.entranceIndex == 0xC020) || (gSaveContext.entranceIndex == 0xC060)) {
+                    this->actionFunc = func_80AD16A8;
+                    return;
+                }
+                if (gSaveContext.entranceIndex == 0xC030) {
+                    func_80AD1398(this);
+                    this->actionFunc = func_80AD1634;
+                    return;
+                }
+                func_80AD144C(this, globalCtx);
+                return;
+            }
+            func_80AD144C(this, globalCtx);
+            return;
+
+        case 1:
+            this->unk_1EC = 0xF;
+            Actor_ChangeAnimation(&this->anime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            return;
+
+        case 2:
+            this->unk_1EC = 0x10;
+            Actor_ChangeAnimation(&this->anime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            return;
+
+        case 3:
+            this->actor.flags &= -2;
+            this->actionFunc = func_80AD16A8;
+            return;
+
+        default:
+            Actor_MarkForDeath(&this->actor);
+            return;
+    }
+}
+
+void EnOsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsn* this = THIS;
+
+    SkelAnime_Free(&this->anime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnOsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    u32 pad1;
+    u16 pad2;
+    u32 sp34;
+    EnOsn* this = THIS;
+
+    sp34 = Flags_GetSwitch(globalCtx, 0);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    SkelAnime_Update(&this->anime);
+    if ((this->actor.params & 3) == 0) {
+        if (sp34 != 0) {
+            this->actor.flags |= 1;
+            func_80AD0830(this, globalCtx);
+            this->actor.draw = EnOsn_Draw;
+        } else {
+            this->actor.draw = 0;
+            this->actor.flags &= -2;
+        }
+    }
+    func_800E9250(globalCtx, &this->actor, (Vec3s*)this->unk_1D8, (Vec3s*)&this->unk_1D8[6], this->actor.focus.pos);
+}
+
+s32 func_80AD1DA8(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor,
+                  Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+
+    if (this->unk_1F0 && limbIndex == 0xB) {
+        Matrix_InsertXRotation_s(this->unk_1DA, 1);
+    }
+    if ((this->unk_1EC == 9 || this->unk_1EC == 8) && limbIndex == 0xA) {
+        *dList = 0;
+    }
+    return 0;
+}
+
+void func_80AD1E28(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
+    Vec3f sp30 = D_80AD2574;
+    Vec3s sp28 = D_80AD2580;
```

Comment:
This kind of assignations usually means the data is inlined
```suggestion
    Vec3f sp30 = { 0.0f, 0.0f, 0.0f };
    Vec3s sp28 = { 0x9920, -0x384, -0x320 };
```

Reply:
Can't get it to match inlined

Reply:
Huh, weird. What's the problem exactly? You remembered to remove the `D_80AD2574` and `D_80AD2580` symbols right?

Reply:
Could you share the diff when you try to move this inline?

Reply:
When I try to move it inline, and run ./diff.py -mwo3 EnOsn_PostLimbDraw, the diff of that function itself still shows a match. But, when I run make, the ROM doesn't match. When I run first_diff.py, I get this:

❯ ./first_diff.py 
First difference at ROM addr 0xF3786C, EnOsn_PostLimbDraw (RAM 0x80AD1E28, ROM 0xF37848, build/src/overlays/actors/ovl_En_Osn/z_en_osn.o)
Bytes: 25:CE:25:88 vs 25:CE:25:74
Instruction difference at ROM addr 0xF37F94, D_80AD2588 (RAM 0x80AD2574, ROM 0xF37F94, build/src/overlays/actors/ovl_En_Osn/z_en_osn.o)
Bytes: 06:01:66:F8 vs 00:00:00:00
Instruction difference at ROM addr 0xF37F98, D_80AD258C (RAM 0x80AD2578, ROM 0xF37F98, build/src/overlays/actors/ovl_En_Osn/z_en_osn.o)
Bytes: 06:01:6E:F8 vs 00:00:00:00
Instruction difference at ROM addr 0xF37F9C, D_80AD2590 (RAM 0x80AD257C, ROM 0xF37F9C, build/src/overlays/actors/ovl_En_Osn/z_en_osn.o)
Bytes: 06:01:76:F8 vs 00:00:00:00
Instruction difference at ROM addr 0xF37FA0, D_80AD2594 (RAM 0x80AD2580, ROM 0xF37FA0, build/src/overlays/actors/ovl_En_Osn/z_en_osn.o)
Bytes: 06:01:7E:F8 vs 99:20:FC:7C
Instruction difference at ROM addr 0xF37FA4, D_80AD2598 (RAM 0x80AD2584, ROM 0xF37FA4, build/src/overlays/actors/ovl_En_Osn/z_en_osn.o)
Bytes: 06:01:82:F8 vs FC:E0:00:00

20 differing word(s).

Reply:
I was thinking more of a diff like this: a screenshot of the diff.py script when it's non-matching

![image](https://user-images.githubusercontent.com/47598039/152681573-067bd70d-592b-47a4-9cd4-4020906f3499.png)

Reply:
When you moved the data inline, did you also move the `TexturePtr`s below `EnOsn_PostLimbDraw`? If i had to guess those were probably in function statics of `EnOsn_Draw`.

Reply:
Ah, that fixed it.